src/parser/grammar.lalrpop

```diff
@@ -298,8 +298,15 @@ Applicative: UniTerm = {
 };
 
 // The parametrized array type.
-TypeArray: Types = "Array" <AsType<RecordOperand>> =>
-    Types(TypeF::Array(Box::new(<>)));
+TypeArray: Types = {
+     <l: @L> "Array" <t: AsType<RecordOperand>>  <r: @R> => {
+        let pos = mk_pos(src_id, l, r);
+        Types {
+            ty: TypeF::Array(Box::new(t)),
+            pos, 
+        }
+    }
+};
```

Comment:
As long as there's a `with_pos` method on `Types`, you can actually use the `WithPos` macro, instead of rewriting those rules. That is, you let this rule unchanged (beside using `with_default_pos` or `from`). Then, in the rules that use `TypeArray`, you replace it with `WithPos<TypeArray>`. Macro are a form of syntactic substitution in LALRPOP, so there's no "typechecking" happening. The advantage is that you don't have to care about `<l: @L>` or `mk_pos` which are contaminating parsing code and rules: everything is offloaded to the `WithPos` macro rule.